### PR TITLE
#4580. Client updates

### PR DESCRIPTION
**Pull request by rmakulov**

### CODE REVIEW

File: atsd_client/_time_utilities.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+
+import dateutil.parser
+import sys
+import time
 from datetime import datetime
 from dateutil.tz import tzlocal, tzutc
-import numbers
+
 try:
     unicode = unicode
```
Comment (raipc) [inline atsd_client/_time_utilities.py]: Let's use `six` instead of this trick

File: atsd_client/models/_meta_models.py [hunk 2]
```diff
@@ -243,16 +309,18 @@ class EntityGroup():
...
         #: `str` group membership expression. The expression is applied to entities to automatically add/remove members of this group
         self.expression = expression
         #: `dict`
         self.tags = tags
+        #: `bool`
+        self.enabled = enabled
     
     def __repr__(self):
-        return "<EntityGroup name={name}, expression={expression}, tags={tags}".format(name=self.name, expression=self.expression, tags=self.tags)
+        return "<EntityGroup name={name}, expression={expression}, tags={tags}".format(name=self.name, expression=self.expression, tags=self.tags, enabled=self.enabled)
```
Comment (raipc) [inline atsd_client/models/_meta_models.py]: Shouldn't there be a closing chevron?

File: atsd_client/_time_utilities.py [hunk 3]
```diff
@@ -58,7 +60,50 @@ def to_iso_utc(time):
...
+    If first date is not specified return sys.maxsize
+    If second date is not specified set it to current time.
+    :param prev_date: None | iso `str` | :class:`datetime` | `int`
+    :param next_date: None | iso `str` | :class:`datetime` | `int`
+    :return: timediff in minutes
+    """
+    if prev_date is None:
+        return sys.maxsize
+
+    if isinstance(prev_date, (str, unicode)):
```
Comment (raipc) [inline atsd_client/_time_utilities.py]: won't work for bytes in python3

**Comment by rodionos** [comment]
log abs path to connection.properties
log connection url and username
no-error accessor to tags.{tag-name}

**Comment by rodionos** [comment]
* Update installation docs (offline) and readme to include 'six' module in requirements
* Replace `http://` and port `8088` with `https://` and port `8443`

File: atsd_client/_time_utilities.py [hunk 4]
```diff
@@ -52,7 +49,7 @@ def to_iso_utc(time):
     aux_time = None
     if time is None:
         aux_time = _current_aware_datetime()
-    if isinstance(time, (str, unicode)):
+    if isinstance(time, (str, six.text_type)):
```
Comment (raipc) [inline atsd_client/_time_utilities.py]: Let's also replace str with six.binary_type

**Comment by rodionos** [comment]
Add get_elapsed_minutes to Entity, Metric, and Series models.

**Comment by rmakulov** [comment]
Add versions for platform-dependent modules.

**Comment by rodionos** [comment]
Update offline installation doc with information on how to download modules for a specific OS (and which command on the target server to run to determine the level/parameter) and a note that for systems earlier than 2.7.x, the download should be performed on a machine with the same (kernel?) level.

File: atsd_client/_jsonutil.py [hunk 5]
```diff
@@ -27,18 +28,23 @@ def serialize(target):
...
         return target
     elif target is None:
         return None
     else:
         try:
             result = {}
-            props = target.__dict__.keys()
-            for prop in props:
-                serialized_property = serialize(target.__dict__[prop])
+            for key, value in vars(target).items():
```
Comment (raipc) [inline atsd_client/_jsonutil.py]: What about this code?

```python
for key, value in six.iteritems(vars(type(target))):
    if isinstance(value, property):
        serialized_property = serialize(getattr(target, key))
        if serialized_property is not None:
            result[key] = serialized_property
return result
```

Comment (rmakulov) [reply atsd_client/_jsonutil.py]: Then I will add properties for query models as well. They do not use getters in current implementation: only setters.

Comment (rodionos) [reply atsd_client/_jsonutil.py]: >properties for query models as well

How much time is this going to require? Is this a big change in terms of lines of code that will require extensive testing?

Comment (rodionos) [reply atsd_client/_jsonutil.py]: As discussed, this change will take a lot of time, with a cascading task to rewrite serialization in multiple models. As a result, this change shall be implemented in a separate branch.

**Pull request merged**